src/Model/Order/Order.php

```diff
@@ -122,20 +171,82 @@ public static function create(
+        $this->businessName = $event->getBusinessName();
         $this->businessAddress = $event->getBusinessAddress();
-        $this->contactPerson   = $event->getContactPerson();
-        $this->isPaid          = $event->getHasPaid();
-        $this->amount          = $event->getAmount();
+        $this->contactPerson = $event->getContactPerson();
+        $this->amount = $event->getAmount();
+        $this->paymentStatus = $event->getPaymentStatus();
+        $this->datePaid = DatePaid::null();
+        $this->paymentAmount = NonNullPaymentAmount::fromNative([
+            'amount' => 0,
+            'currency' => 'gbp'
+        ]);
+        $this->paymentStatus = NonNullPaymentStatus::UNPAID();
     }
+
+    public function pay(): void
+    {
+        $this->recordThat(
+            OrderWasPaid::occur($this->getId()->toNative())
+        );
+    }
+
+    private function applyOrderWasPaid(OrderWasPaid $event): void
+    {
+        $this->id = $event->getId();
+        $this->datePaid = DatePaid::fromNative($event->createdAt()->format(DATE_RFC3339));
+    }
+
+    public function receivePayment(
+        NonNullEntityId $id,
+        NonNullPayeeName $payeeName,
+        NonNullPaymentAmount $paymentAmount,
+        NonNullPaymentType $paymentType
+    ): void {
+        if($paymentAmount->getMoney()->isZero()) {
+            throw new PaymentAmountMustBeGreaterThanZero($paymentAmount);
+        }
+        if($paymentAmount->getMoney()->isNegative()) {
+            throw new PaymentAmountMustNotBeNegative($paymentAmount);
+        }
+        if($this->getPaymentAmount()->getMoney()->add($paymentAmount->getMoney())->greaterThan($this->getAmount()->getMoney())) {
+            throw new PaymentAmountMustNotExceedTotalOrderAmount($paymentAmount);
+        }
+        $this->recordThat(
+            PaymentWasReceived::occur(
+                $this->getId()->toNative(),
+                [
+                    'paymentId' => $id->toNative(),
+                    'payeeName' => $payeeName->toNative(),
+                    'amount' => $paymentAmount->toNative(),
+                    'type' => $paymentType->toNative(),
+                ]
+            )
+        );
+    }
+
+    private function applyPaymentWasReceived(PaymentWasReceived $event): void {
+        $this->paymentID = $event->getPaymentId();
+        $this->paymentAmount = new NonNullPaymentAmount($this->getPaymentAmount()->getMoney()->add($event->getAmount()->getMoney()));
+        if($this->getPaymentAmount()->getMoney()->equals($this->getAmount()->getMoney())) {
+            $this->paymentStatus = NonNullPaymentStatus::PAID();
+        }
+        if($this->getPaymentAmount()->getMoney()->lessThan($this->getAmount()->getMoney())) {
+            $this->paymentStatus = NonNullPaymentStatus::PART_PAID();
+        }
+        $this->payeeName = $event->getPayeeName();
+        $this->paymentDate = PaymentDate::fromNative($event->createdAt()->format('Y-m-d'));
```

Comment:
Can't remember, but would they ever want to mark a payment as being in the past?

Reply:
Yes, it has been removed now.